src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -114,11 +117,19 @@ public JobService(IamService samService,
     public void initialize() {
         try {
             List<String> recordedStairways;
-            migrate.migrateDatabase();
+            // check if we should allow DropAllOnStart
+
+            boolean allowDropAllOnStart = migrate.allowDropAllOnStart();
+            boolean dropAllOnStartConfig = migrateConfiguration.getDropAllOnStart();
+            boolean dropAllOnStart = allowDropAllOnStart && dropAllOnStartConfig;
+            logger.info("DropAllOnStart? {} => allowDropAllOnStart = {} && dropAllOnStartConfig = {}",
+                dropAllOnStart, allowDropAllOnStart, dropAllOnStartConfig);
+
+            migrate.migrateDatabase(dropAllOnStart);
```

Comment:
Moving this check here so that we perform it before running migrations on BOTH datarepo and stairway.